0000-interledger-quoting-protocol-v2.md

```diff
@@ -0,0 +1,87 @@
+3. **Recognition of Quoting as an Application-Specific Concern** - Given the unreliability of quotes, different applications will need to handle fluctuating rates and liquidity differently. For example, the mechanism used to judge the cost of a stream of payments versus a single payment must actively take into account the changing rates. This suggests that different applications may need different types of quoting functionality or may not use quoting at all.
+4. **Enabling Evolution of the Quoting Protocol** - Protocols on the Interledger Layer must be implemented by all participants and will thus be more difficult to update. Taking quoting out of the realm of connectors will enable different quoting protocols to evolve over time.
+
+### Scope
+
+ILQPv2 is intended for senders to determine the rate between the source and destination amounts for a payment.
+
+It is primarily designed for individual payments. While it may be used for streams of payments, other protocols that actively take into account exchange rates changing over time may be better suited for such use cases.
+
+### Operation
+
+Senders may use a dedicated quoting module or ILQPv2 may be included in a Transport Layer protocol. A sender's ILQPv2 module is responsible for initiating specially crafted test payments to receivers. The receiver's ILQPv2 module is responsible for handling the incoming test payment and rejecting it with a specific error message.
+
+## Overview
+
+### Relation to Other Protocols
+
+ILQPv2 uses payments sent with the Interledger Protocol. In order for this protocol to work, Interledger connectors MUST NOT reject payments where the amount in the ILP Packet exceeds what the connector thinks should be delivered to the receiver based on the accompanying transfer amount. Connectors SHOULD forward the payment and apply their normal exchange rate. Connectors MUST also relay error messages back to the sender.
+
+ILQPv2 is intended to be used alongside Transport Layer protocols. It uses a transfer condition with a special prefix (`ffffffffffffff` in hexadecimal or `__________` in base64url encoding). Other end-to-end protocols SHOULD ignore to avoid interfering with ILQPv2 and other protocols that use well-known conditions.
+
+## Model of Operation
+
+### Fixed Source Amount
+
+This details the steps for a sender to get a quote for a fixed source amount ("if I send this much, how much will the receiver get?"):
+
+1. The sender constructs an ILP packet with the receiver's ILP Address, an amount equal to the maximum value for unsigned 64-bit integers (18446744073709551615), and no data.
```

Comment:
would it be theoretically possible that the money actually gets reserved or the receiver redeems?

Reply:
Nope. Connectors use the transfer amount, not the amount in the ILP packet to determine how much they're going to put on hold. And by using a special hash that no one knows the preimage for, it's impossible for the payment to be redeemed.